lib/services/badges_S/badge_service.dart

```diff
@@ -15,6 +17,37 @@ class BadgeService {
   final FirebaseFirestore _firestore = FirebaseFirestore.instance;
   final FirebaseAuth _auth = FirebaseAuth.instance;
 
+  Future<List<String>> getConnectionsBadgeStatus(String badgeName) async {
+  final currentUser = _auth.currentUser?.uid;
+
+  List<String> connections = await ConnectionService().getConnections("connections", currentUser);
+  List<String> unlockedProfiles = []; 
+
+  for (String connectionId in connections) {
+    try {
+      DocumentSnapshot<Map<String, dynamic>> badgeSnapshot =
+          await _firestore.collection('badges').doc(connectionId).get();
+
+      if (badgeSnapshot.exists && badgeSnapshot.data() != null) {
+        Map<String, dynamic> badgeData = badgeSnapshot.data()!;
+
+        if (badgeData.containsKey(badgeName) && badgeData[badgeName]['unlocked'] == true) {
+          StorageService storageService = StorageService();
+          String profilePictureUrl =
+              await storageService.getProfilePictureUrl(currentUser!);
```

Comment:
The one thing I saw when I ran it on my side, if someone has unlocked the badge, that profile has my profile picture. I believe this should be an easy fix thouh , if you just replace currentUser with the connection's uid ( I think you named it ConnectionID? This should fix the profile pictures on the badge pages.

Reply:
Ah thank you.  I must have missed that...Thank you for pointing it out